authz/providers/azure/rbac/checkaccessreqhelper.go

```diff
@@ -324,7 +326,11 @@ func ConvertCheckAccessResponse(body []byte) (*authzv1beta1.SubjectAccessReviewS
 
 	if strings.ToLower(response[0].Decision) == Allowed {
 		allowed = true
-		verdict = AccessAllowedVerdict
+		if (AzureRoleAssignment{} != response[0].AzureRoleAssignment) && (response[0].AzureRoleAssignment.Id != "" || response[0].AzureRoleAssignment.RoleDefinitionId != "" || response[0].AzureRoleAssignment.PrincipalId != "") {
```

Comment:
should this be empty check or nil check?

Reply:
empty as while unmarshalling , those fields will never be null , they will be empty as they are of string type

Reply:
We are fine with string having empty. What we want to make sure is we don't crash. Althogh Id is string type they are part of AzureRoleAssignment which is struct that can be nil. right?

Reply:
AzureRoleAssignment will not be nil, as json unmarshal will make it as an empty struct

Reply:
got it. Then let's have only one message. In worst case we would print those fields as empty. Anyways it's not expected to be empty on most of the cases.